hera_cal/abscal.py

```diff
@@ -195,14 +202,45 @@ def __init__(self, model, data, refant=None, wgts=None, antpos=None, freqs=None,
         self.bls = None
         if self.antpos is not None:
             # center antpos about reference antenna
-            self.antpos = odict(map(lambda k: (k, antpos[k] - antpos[self.refant]), antpos.keys()))
+            self.antpos = odict(map(lambda k: (k, antpos[k] - antpos[self.refant]), self.ants))
             self.bls = odict([(x, self.antpos[x[0]] - self.antpos[x[1]]) for x in self.keys])
             self.antpos_arr = np.array(map(lambda x: self.antpos[x], self.ants))
             self.antpos_arr -= np.median(self.antpos_arr, axis=0)
 
         # setup gain solution keys
         self._gain_keys = map(lambda p: map(lambda a: (a, p), self.ants), self.gain_pols)
 
+        # perform baseline cut
```

Comment:
For namespace clarity and future utility, I'd break this out into its own function.

Reply:
ok!